pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -272,6 +274,23 @@ func (c *Controller) syncPodAutoscaler(ctx context.Context, key, ns, name string
 	// Reaching this point, we had an error in processing, clearing up global error
 	podAutoscalerInternal.SetError(nil)
 
+	// Check that targetRef is not set to the cluster agent
+	clusterAgentRef := autoscalingv2.CrossVersionObjectReference{
```

Comment:
You also need to check autoscaler namespace vs cluster agent namespace: https://github.com/DataDog/datadog-agent/blob/main/pkg/util/kubernetes/apiserver/common/common.go#L40-L49

Also not sure you need to compare the full `CrossVersionObjectReference`. Basically if something has some name and namespace than DCA it's suspicious enough.

Reply:
Thinking more about it, you should try to cover both Cluster Agent Deployment AND direct ReplicaSet